matlab/src/ObjectPrx.cpp

```diff
@@ -306,59 +284,23 @@ extern "C"
 
         try
         {
-            Ice::Context ctx;
-            getContext(context, ctx);
-            function<void()> token = proxy->ice_invokeAsync(
-                op,
-                mode,
-                params,
-                [proxy, f](bool ok, pair<const byte*, const byte*> outParams)
-                { f->finished(proxy->ice_getCommunicator(), proxy->ice_getEncodingVersion(), ok, outParams); },
-                [f](exception_ptr e) { f->exception(e); },
-                [f](bool /*sentSynchronously*/) { f->sent(); },
-                ctx);
-            f->token(token);
-            *future = new shared_ptr<InvocationFuture>(move(f));
-        }
-        catch (...)
-        {
-            return convertException(std::current_exception());
-        }
-        return 0;
-    }
-
-    mxArray* Ice_ObjectPrx_ice_invokeAsyncNC(
-        void* self,
-        const char* op,
-        int m,
-        mxArray* inParams,
-        unsigned int size,
-        void** future)
-    {
-        const auto proxy = restoreProxy(self);
-        pair<const byte*, const byte*> params(0, 0);
-        if (!mxIsEmpty(inParams))
-        {
-            params.first = reinterpret_cast<byte*>(mxGetData(inParams));
-            params.second = params.first + size;
-        }
-        auto mode = static_cast<Ice::OperationMode>(m);
-
-        *future = 0;
-        auto f = make_shared<InvocationFuture>(
-            proxy->ice_isTwoway(),
-            proxy->ice_isBatchOneway() || proxy->ice_isBatchDatagram());
+            const Ice::Context* ctxPtr = &Ice::noExplicitContext;
+            Ice::Context cppContext;
+            if (!mxIsEmpty(context))
+            {
+                getContext(context, cppContext);
+                ctxPtr = &cppContext;
+            }
 
-        try
-        {
             function<void()> token = proxy->ice_invokeAsync(
                 op,
                 mode,
                 params,
                 [proxy, f](bool ok, pair<const byte*, const byte*> outParams)
                 { f->finished(proxy->ice_getCommunicator(), proxy->ice_getEncodingVersion(), ok, outParams); },
                 [f](exception_ptr e) { f->exception(e); },
-                [f](bool /*sentSynchronously*/) { f->sent(); });
+                [f](bool /*sentSynchronously*/) { f->sent(); },
+                *ctxPtr);
```

Comment:
git-diff gets confused here: I removed the NC function and updated the "with context" function.